pkg/karmadactl/register/register.go

```diff
@@ -196,6 +198,9 @@ func NewCmdRegister(parentCommand string) *cobra.Command {
 	flags.Int32Var(&opts.CertExpirationSeconds, "cert-expiration-seconds", DefaultCertExpirationSeconds, "The expiration time of certificate.")
 	flags.BoolVar(&opts.DryRun, "dry-run", false, "Run the command in dry-run mode, without making any server requests.")
 	flags.StringVar(&opts.ProxyServerAddress, "proxy-server-address", "", "Address of the proxy server that is used to proxy to the cluster.")
+	flags.StringVar(&opts.FeatureGates, "feature-gates", "", ""+
+		"A set of key=value pairs that describe feature gates for alpha/experimental features. "+
+		"Options are:\n"+strings.Join(features.FeatureGate.KnownFeatures(), "\n"))
```

Comment:
```suggestion
	features.FeatureGate.AddFlag(flags)
```
Can we do that this way, which is the way we did at all the other components, like https://github.com/karmada-io/karmada/blob/6144ea5a9b2fbd8bfd33e651d0901842a9f759de/cmd/controller-manager/app/options/options.go#L235

Reply:
I've tried this way, but ran into an issue where I couldn't get the string value of the feature gates when injecting them into the agent's manifest.

Reply:
why?

Reply:
It is worth noting that, you don't need to introduce an additional field into `CommandRegisterOption` anymore.
```
	FeatureGates string
```

Reply:
> why?

https://github.com/karmada-io/karmada/blob/7112723f3de7ba640336202f19dd818e2ca393ca/vendor/k8s.io/component-base/featuregate/feature_gate.go#L122-L153

`features.FeatureGate.AddFlag(flags)` has no method to get the string value of the feature gate. Hard to complete `fmt.Sprintf(‘--feature-gates=%s’, o.FeatureGates)`.